src/util/OAuth2Util.js

```diff
@@ -16,14 +16,21 @@ import Errors from './Errors';
 import Util from './Util';
 const util = {};
 
-function hasResponseType (params, type) {
-  if (_.isArray(params.responseType)) {
-    return _.contains(params.responseType, type);
+function hasResponseType (responseType, type) {
+  if (_.isArray(responseType)) {
+    return _.contains(responseType, type);
   } else {
-    return type === params.responseType;
+    return type === responseType;
   }
 }
 
+util.isAuthorizationCodeFlow = function (settings) {
+  const authClient = settings.getAuthClient();
+  const pkce = authClient.options.pkce;
+  const responseType = settings.get('authParams.responseType');
```

Comment:
It's controversial that `authParams.responseType` has been removed from `props` definition but still being used.

Reply:
The `authParams` section is still supported, as it is used to configure the "internal" instance. However, the backbone props definitions potentially conflict with the actual configuration definition, which is defined here: https://github.com/okta/okta-auth-js/blob/master/lib/types/OktaAuthOptions.ts

This object is passed in blindly to the constructor of OktaAuth. The reason I am reading responseType here from widget settings rather than the authClient.options is because `responseType` is not saved when passed on the OktaAuth constructor, it is a `getToken` option: https://github.com/okta/okta-auth-js/blob/master/lib/types/api.ts#L77

I think **maybe** this is a bug in AuthJS, we should take `responseType` in the constructor, but also allow it to be overridden on a call to `getToken`. It would need to be added to the list here: https://github.com/okta/okta-auth-js/blob/master/lib/browser/browser.ts#L129

Reply:
https://oktainc.atlassian.net/browse/OKTA-340725 I am going to try to get this in before the 4.1 release which is planned for later this week. Then we can update SIW to 4.1 and it will not need to check `authParams.responseType` from settings.